lib/internal/modules/esm/resolve.js

```diff
@@ -937,6 +938,15 @@ function checkIfDisallowedImport(specifier, parsed, parsedParentURL) {
   }
 }
 
+function throwIfInvalidParentURL(parentURL) {
+  if (parentURL === undefined) {
+    return; // Main entry point, so no parent
+  }
+  if (typeof parentURL !== 'string' && !isURLInstance(parentURL)) {
+    throw new ERR_INVALID_ARG_TYPE('parentURL', ['string', 'URL'], parentURL);
+  }
+}
```

Comment:
Is this a user-input validation util? (I can't think how else an invalid parentURL would be possible—aside from a bug in node)

Reply:
The context is user provided when there are custom hooks.